recipe/conda_build_config.yaml

```diff
@@ -492,6 +492,8 @@ mumps_mpi:
   - 5.2
 mumps_seq:
   - 5.2
+nccl:
+  - 2.4.6.1
```

Comment:
This must match the pinning `x`. For example if the pinning was `x.x`, then this should be `2.4`. Since the pinning is `x` it should be `2`. This is so that downstream packages can use newer features introduced in later versions.

Reply:
Our goal here is to ensure we build against the oldest version of `nccl` we care to have supported by all packages using it for maximum compatibility.

If we wanted to bump this later, we can always do a migrator, which would be explicit and preferable to letting this creep forward implicitly due to when we happened to rebuild downstream dependencies.

Reply:
cc @mingwandroid

Reply:
FWIW there are other libraries doing this currently. Please see `kealib` as one such example, which only requires [`x.x` compatibility]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/0df9ddae9c1f6a8f862795a8eee324544e771512/recipe/conda_build_config.yaml#L204-L205 ), but [sets the version used in `host` to `1.4.10`]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/0df9ddae9c1f6a8f862795a8eee324544e771512/recipe/conda_build_config.yaml#L427-L428 ).